src/commands/queue.ts

```diff
@@ -17,6 +17,22 @@ export function addQueueCommands(bot: DiscordBot, config: BotConfig, daos: Daos)
   const ytApi = new YoutubeAPI(config.tokens.youtube);
   const scApi = new SoundCloudAPI(config.tokens.soundcloud);
 
+  const add = async (message: Message, params: string[]) => {
+    try {
+      if (params.length === 0) return await message.reply("You didn't specify what to add!");
+      const queryResults = await Utils.songQuery(message, params.join(' '), scUsers, users, scApi, ytApi);
+      if (_.isNil(queryResults)) return;
+      await queuePlayerManager.get(message.guild.id).enqueue(queryResults.songs, queryResults.nextFlag);
```

Comment:
There's just a couple edge cases with this approach still - if any error occurs during the `add` logic before we enqueue then we still would have already cleared the previous queue. Would be best to only clear the queue and add to the queue after the new songs have been successfully fetched.

Ideally should refactor to support this but also I would be okay if you just want to go back to having the duplicate logic in both `add` and `replace` and simply have the line at `24.5` to clear the queue first in the `replace` logic - also update responses to indicate that queue was replaced.

Reply:
Renamed the function to `insertToQueue` with a mode parameter that offers both add and replace functionality. Checks the mode to determine what to do. I used enum so it should be easily extendable. `add` and `replace` calls with a specific mode that directs `insertToQueue` into either adding to or replacing the queue.